### PR TITLE
`object_lifetime_tracker_act` Code and Tests

### PR DESCRIPTION
**Pull request by parth21999**

### CODE REVIEW

File: tests/object_lifetime_tracker_ut/object_lifetime_tracker_ut.c [hunk 1]
```diff
@@ -1202,4 +1229,217 @@ TEST_FUNCTION(object_lifetime_tracker_destroy_all_objects_for_key_key_not_found)
...
+    // assert
+    ASSERT_ARE_EQUAL(OBJECT_LIFETIME_TRACKER_ACT_RESULT, OBJECT_LIFETIME_TRACKER_ACT_ERROR, result);
+    ASSERT_ARE_EQUAL(char_ptr, umock_c_get_expected_calls(), umock_c_get_actual_calls());
+
+    // cleanup
+    object_lifetime_tracker_destroy(object_lifetime_tracker);
+}
+
+/*Tests_SRS_OBJECT_LIFETIME_TRACKER_43_083: [ If object is NULL, object_lifetime_tracker_act shall fail and return OBJECT_LIFETIME_TRACKER_ACT_ERROR. ]*/
+TEST_FUNCTION(object_lifetime_tracker_act_with_NULL_OBJECT_FAILS)
```
Comment (anporumb) [inline tests/object_lifetime_tracker_ut/object_lifetime_tracker_ut.c:1265]: >OBJECT_FAILS [](http://example.com/codeflow?start=52&length=12)

consider using non screaming capitalization

File: tests/object_lifetime_tracker_ut/object_lifetime_tracker_ut.c [hunk 2]
```diff
@@ -1202,4 +1229,217 @@ TEST_FUNCTION(object_lifetime_tracker_destroy_all_objects_for_key_key_not_found)
...
+
+            // assert
+            ASSERT_ARE_EQUAL(OBJECT_LIFETIME_TRACKER_ACT_RESULT, OBJECT_LIFETIME_TRACKER_ACT_ERROR, result);
+        }
+    }
+
+    // cleanup
+    ASSERT_ARE_EQUAL(OBJECT_LIFETIME_TRACKER_UNREGISTER_OBJECT_RESULT, OBJECT_LIFETIME_TRACKER_UNREGISTER_OBJECT_OK, object_lifetime_tracker_unregister_object(object_lifetime_tracker, test_key_1, test_object_1));
+    object_lifetime_tracker_destroy(object_lifetime_tracker);
+}
```
Comment (anporumb) [inline tests/object_lifetime_tracker_ut/object_lifetime_tracker_ut.c:1443]: These tests do not yield their maximum potential for coverage. For example, putting a breakpoint in line 490 - it is never hit, therefore line 490 of object_lifetime_tracker.c is not tested.

Other lines where breakpoints are not hit are: 509 and 528.

**Review by anporumb — APPROVED** [review]
:shipit:

**Review by dcristoloveanu — APPROVED** [review]
:shipit:

File: tests/object_lifetime_tracker_ut/object_lifetime_tracker_ut.c (hunk 1 above)
Comment (parth21999) [reply tests/object_lifetime_tracker_ut/object_lifetime_tracker_ut.c:1265]: Done

File: tests/object_lifetime_tracker_ut/object_lifetime_tracker_ut.c (hunk 2 above)
Comment (parth21999) [reply tests/object_lifetime_tracker_ut/object_lifetime_tracker_ut.c:1443]: Fixed. I forgot to set up the expectations for the negative test

**Review by dcristoloveanu — APPROVED** [review]
:shipit:

**Pull request merged**